moveit_py/src/moveit_py/moveit_py_utils/include/moveit_py/moveit_py_utils/ros_msg_typecasters.h

```diff
@@ -63,15 +63,15 @@ struct RosMsgTypeCaster
                                 serialized_msg.get_rcl_serialized_message().buffer_length);
 
     // get Python object type
-    const std::string ros_msg_name = rosidl_generator_traits::data_type<T>();
+    const std::string ros_msg_name = rosidl_generator_traits::name<T>();
 
-    // find delimiting '::' in ros_msg_name
-    std::size_t pos1 = ros_msg_name.find("::");
-    std::size_t pos2 = ros_msg_name.find("::", pos1 + 2);
+    // find delimiting '/' in ros_msg_name
+    std::size_t pos1 = ros_msg_name.find("/");
+    std::size_t pos2 = ros_msg_name.find("/", pos1 + 1);
     py::module m = py::module::import((ros_msg_name.substr(0, pos1) + ".msg").c_str());
```

Comment:
Please replace this code with my suggestion. Extracting and concatenating std::string requires several copy operations, while my code only copies once. As the type conversion code will be frequently used, we should care about this optimization.

Reply:
I will update this commit with a rebase now.

Reply:
Apologies for the messy commit history. I'd be happy reformat this history into something more reasonable before a merge.

Reply:
[58a541f](https://github.com/ros-planning/moveit2/pull/1546/commits/58a541fbf78c0f492de267af15597b03ee181ed5)